File: app/ecs/systems/ControllerSystem.lua
local ____exports = {}
local ____Controller = require("game.core.input.Controller")
local newController = ____Controller.newController
local ____ECS = require("ecs.core.ECS")
local ecs = ____ECS.ecs
function ____exports.ControllerSystem()
    local controllers = {}
    return ecs.systems.create(
        {"InputComponent"},
        {input = function(____, entities, action_id, action)
            for ____, e in ipairs(entities) do
                if controllers[e.id] == nil then
                    controllers[e.id] = newController()
                    e.components.InputComponent:init(controllers[e.id])
                end
                controllers[e.id].input(action_id, action)
            end
        end}
    )
end
return ____exports

File: src/ecs/systems/ControllerSystem.ts

import { newController } from "../../game/core/input/Controller";
import { ecs } from "../core/ECS";

export function ControllerSystem() {
    const controllers: { [id: number]: ReturnType<typeof newController> } = {};
    return ecs.systems.create(['InputComponent'], {
        input: (entities, action_id, action) => {
            for (const e of entities) {
                if (controllers[e.id] == undefined) {
                    controllers[e.id] = newController();
                    e.components.InputComponent.init(controllers[e.id]);
                }
                controllers[e.id].input(action_id, action);
            }
        }
    });
}